plutus-tx-plugin/test/Budget/9.6/andWithLocal.uplc.golden

```diff
@@ -0,0 +1,12 @@
+program
+  1.1.0
+  ((\x y ->
+      (\lessThanInteger ->
+         (\l ->
+            (\r -> force (case l [(delay r), (delay (constr 1 []))]))
+              (lessThanInteger y 3))
+           (lessThanInteger x 3))
+        (\x y ->
+           force ifThenElse (lessThanInteger x y) (constr 0 []) (constr 1 [])))
```

Comment:
I was checking if this term and the "local" one are equivalent, and noticed something weird. It looks like we're overloading `lessThanInteger` here:
- on line 4 it's a bound variable
- on line 10 (outside ^'s scope) it's, I would guess, a builtin

I think the two terms (this and the "local" one) are alpha-equivalent, though.

Reply:
They are not equivalent. The one introduced in line 4 corresponds to the one in `andWithLocal.pir.golden`, line 5. It returns the non-builtin `Bool` (which in PIR is just `Bool`, and in UPLC it is `constr 0 []` for True and `constr 1 []` for False). And it comes from `PlutusTx.Builtins.lessThanInteger`. In the `WithGhcOpts` test, it is inlined by GHC.

The `lessThanInteger` in line 10 is indeed the builtin function, and it returns the builtin `Bool`. The condition in the builtin `ifThenElse` requires the builtin `Bool`.

Reply:
Yes, there is name shadowing here, unfortunately.